js/gemini.js

```diff
@@ -810,6 +831,39 @@ module.exports = class gemini extends Exchange {
         };
     }
 
+    async fetchDepositAddressesByNetwork (code, params = {}) {
+        await this.loadMarkets ();
+        const network = this.safeString (params, 'network');
+        if (!network) {
+            throw new BadRequest ('fetchDepositAddressByNetwork requires a network parameter');
+        }
+        params = this.omit (params, 'network');
+        let networkCode = undefined;
+        let networkId = undefined;
+        const networks = this.safeValue (this.options, 'networks', {});
+        const networkIds = this.safeValue (this.options, 'networkIds', {});
+        networkId = this.safeString (networks, network);
+        networkCode = this.safeString (networkIds, networkId);
+        if (networkId === undefined) {
+            networkCode = this.safeString (networkIds, network);
+            networkId = network;
+        }
+        const request = {};
+        request['network'] = networkId;
+        const response = await this.privatePostV1AddressesNetwork (this.extend (request, params));
+        const addressList = [];
+        for (let i = 0; i < response.length; i++) {
+            const entry = response[i];
+            const address = this.safeString (entry, 'address');
+            addressList.push (address);
+        }
+        return {
```

Comment:
We will need to return the following type of structure from this method:

```JavaScript
{
    'BTC': [ // unified network code
        // a list of unified deposit address structures
        // from https://github.com/ccxt/ccxt/wiki/Manual#address-structure
        {
            'currency': currency, // currency code
            'network': 'BTC',     // a list of deposit/withdraw networks, ERC20, TRC20, BSC20 (see below)
            'address': address,   // address in terms of requested currency
            'tag': tag,           // tag / memo / paymentId for particular currencies (XRP, XMR, ...)
            'info': response,     // raw unparsed data as returned from the exchange
        },
        {
            // ...
        },
        {
            // ...
        },
        // ...
    ],
}
```

I would suggest to split the depositAddress method to a standalone parser (`parseDepositAddress`, see how it is done in the other exchanges, fill as many fields as we can there). Then use `parseDepositAddresses()` to parse the results. And then `return this.groupBy (results, 'network')`. We can extend the params and the functionality of `parseDepositAddresses()` to support currency codes and networks overrides if necessary.

Reply:
That is, to keep this response format closer to the other exchanges.